### PR TITLE
STV last round use "Final Round Surplus" rather than inactive

### LINKED ISSUE #855: Add 'Final Round Surplus' row in summary.csv
**RCTab Version**: 1.3.999
Config [vmsg surplus transfer test.json](https://github.com/user-attachments/files/15888521/vmsg.surplus.transfer.test.json)
CVR [vmwg cvr.xlsx](https://github.com/user-attachments/files/15888524/vmwg.cvr.xlsx)

In STV, when the final candidate(s) are elected we have a final transfer round to remove any votes over the threshold that are unneeded to elect that candidate. This step ensures that all elected candidates are shown with the same vote total in the final round (otherwise, the last candidate(s) to be elected would show a higher vote total than candidates elected before them). Currently, those surplus votes over the threshold are moved to the Inactive Ballots bucket in the final transfer round. We should make a new bucket, `Final Round Surplus`, under Residual Surplus, and put any surplus votes unneeded to elect in there. This would show up in both summary.csv and summary_extended.csv

I added an example below. The sum of everything in the `a` box should equal `b`. Only the final round where surplus transfers are made from the last elected candidate(s) should have a number for  Final Round Surplus, others should be empty.  
 

Image

### PR DESCRIPTION
**Pull request by artoonie**
There are a few options for implementation here.

I chose to treat the final round surplus similar to how we treat inactive ballots, rather than how we treat residual surplus. Residual surplus is treated as votes that sort of "disappear," showing up in the transfer list but not elsewhere. In this case, we want the final round to not make these votes disappear, but rather, land in a non-candidate and non-inactive bucket.

I played around with other ideas but this felt both the most correct and led to the most natural implementation.

Closes #855

### CODE REVIEW

**Comment by yezr** [comment]
### Testing
- Confirmed for all the races that had `summary.csv` changes that the new value in `Final Round Transfer` exactly matched the difference that no longer went to other inactive buckets

Need to figure out what is up with src/test/resources/network/brightspots/rcv/test_data/test_set_allow_only_one_winner_per_round/test_set_allow_only_one_winner_per_round_expected_summary.json issue during code review

File: src/main/java/network/brightspots/rcv/ResultsWriter.java
Comment (alyssahursh) [inline src/main/java/network/brightspots/rcv/ResultsWriter.java:1031]: There's something about this code (both the old and the new) that has me scratching my head. What would you think about moving these "key" strings into the `StatusForRound` enum as, say, `jsonKey`? Then the logic here might look like this:

```
private Map<String, BigDecimal> getInactiveJsonMap(RoundTally roundTally) {
    List<StatusForRound> statusesToPrint = new ArrayList<>();
    statusesToPrint.add(INVALIDATED_BY_OVERVOTE);
    statusesToPrint.add(INVALIDATED_BY_SKIPPED_RANKING);
    statusesToPrint.add(EXHAUSTED_CHOICE);
    statusesToPrint.add(INVALIDATED_BY_REPEATED_RANKING);

   if (config.usesSurpluses() && roundTally.getRoundNumber() == numRounds) {
      statusesToPrint.add(FINAL_ROUND_SURPLUS);
    }

    return statusesToPrint.stream()
        .collect(Collectors.toMap(StatusForRound::jsonKey, status -> roundTally.getBallotStatusTally(status)));
}
```

**Review by alyssahursh — APPROVED** [review]
Nice to meet you today, Armin! Sorry to drop so abruptly without a proper goodbye. I had a few extra comments for you on this PR. If you find them useful, great! If not, that's fine too :)

**Review by yezr — APPROVED** [review]
LGTM. All code review comments addressed
- `StatusForRound` enum now includes single source of truth for which ones are active/inactive and their `audit.log` text
- logic for whether or not the configuration of a contest will require a Final Round Surplus is pushed up into the `ContestConfig`
- active/inactive count in `Summary` fixed to properly account for `FINAL_ROUND_TRANSFER`

**Comment by yezr** [comment]
I approved thinking that I had seen all of Alyssa's comments but for some reason some of them were not showing when I was looking at the diff of just the two commits since our code review. The ones I weren't seeing were the last two that were suggestions. I thought the only outstanding PR conversations were simply comments on how the updates fixed exactly what we had talked about in the code review. Since those two are suggestions for updates and not requirements for changes I will leave it to Armin to resolve those

I used the shift + click to select the last two commits but it didn't two of Alyssa's four comments were not visible in that diff screen. Weird...
image

File: src/main/java/network/brightspots/rcv/ResultsWriter.java
Comment (artoonie) [reply src/main/java/network/brightspots/rcv/ResultsWriter.java:1031]: Note that we have different strings for different outputs -- the JSON file uses camelCase and the CSV uses Title Plain Text (with spaces). I fear that this difference doesn't belong inside StatusForRound itself (e.g. `plaintextString` vs `camelCaseString`), though could see an argument for standardizing the text across the two formats. That would have a lot of downstream side-effects. Thoughts?

File: src/main/java/network/brightspots/rcv/Tabulator.java
Comment (artoonie) [reply src/main/java/network/brightspots/rcv/Tabulator.java]: Done! And made the subsequent line `final` as well to match.

File: src/main/java/network/brightspots/rcv/ResultsWriter.java
Comment (yezr) [reply src/main/java/network/brightspots/rcv/ResultsWriter.java:1031]: Though #856 doesn't require different strings, I could see other future output formats possibly requiring other strings to match to each of the `StatusForRound` enums. The `.json` vs `.csv` formats are similar enough that it seems silly to have both, but other formats could be different enough and this spot seems as good a place as any.

Comment (alyssahursh) [reply src/main/java/network/brightspots/rcv/ResultsWriter.java:1031]: @artoonie You have more experience in this code than I do, so I'll certainly defer to your judgment. I'm just running off intuition here, and my intuition says that all of the different ways of titling these statuses should be held somewhere in common rather than spread around in different files. My concern is that by spreading these different strings around, we're likely to encourage a third or fourth formatting standard to crop up unnecessarily. But my familiarity with this code base is very thin, and I'm not here to block you!

Comment (artoonie) [reply src/main/java/network/brightspots/rcv/ResultsWriter.java]: Got it! I tried it out and I do think it's cleaner as you suggested, though we previously had three different string types not two:
- camel case, brief (e.g. "overvote")
- title case, using "invalidated" (e.g. "Invalidated by Overvote")
- title case, using "inactive" (e.g. "Inactive by Overvote")

The "invalidated" phrase was only used in the logs. @yezr, I don't recall if that was an intentional choice to use it in the audit logs but not in the output CSV. Are you okay using "inactive" for both, or shall I add a third string to the enum in the last commit here?

Comment (yezr) [reply src/main/java/network/brightspots/rcv/ResultsWriter.java]: Using "inactive" for both will work.

Comment (artoonie) [reply src/main/java/network/brightspots/rcv/ResultsWriter.java]: Great, then I'd like a final review on the last commit from Alyssa then I think we're ready to merge all three?

Thanks for the great feedback, Alyssa! This process has been super helpful.

File: src/main/java/network/brightspots/rcv/ResultsWriter.java
Comment (alyssahursh) [inline src/main/java/network/brightspots/rcv/ResultsWriter.java:1038]: I see that you didn't love the stream suggestion! Since you're using a modifiable map here, you could use the static final immutable list described above like this:
```
private Map<String, BigDecimal> getInactiveJsonMap(RoundTally roundTally) {
    Map<String, BigDecimal> result = new HashMap<>();
    for (StatusForRound statusToPrint : STATUSES_TO_PRINT) {
        result.put(statusToPrint.getCamelCaseKey(), roundTally.getBallotStatusTally(statusToPrint));
    }

    if (config.usesSurpluses() && roundTally.getRoundNumber() == numRounds) {
        result.put(FINAL_ROUND_SURPLUS.getCamelCaseKey(), roundTally.getBallotStatusTally(FINAL_ROUND_SURPLUS));
    }

    return result;
}
```

Again, not asking for changes in this PR! Just pointing out what I see :) If it's useful to you, great! Overall, I think the changes that have already been made are a step in the right direction.

**Review by alyssahursh — COMMENTED** [review]
I know that I was able to approve prior versions, but I somehow can't approve today ("Only users with explicit access to this repository may approve pull requests.") I left two more comments, but I'm not requesting changes. Consider this comment my approval!

**Comment by yezr** [comment]
> I know that I was able to approve prior versions, but I somehow can't approve today ("Only users with explicit access to this repository may approve pull requests.") I left two more comments, but I'm not requesting changes. Consider this comment my approval!

I got to get the approval stuff figured out. Right now anyone can approve and that allows a merge. We don't have a more intermediate "I am done reviewing all the changes" that still allows a later "official" approval for the merge to make its way to `dev`. For now if you can just comment in the PR like this to flag that a review is finished. 

The update to make this list `final` since then LGTM. Approved

**Pull request merged**
